src/sqlfluff/rules/L068.py

```diff
@@ -0,0 +1,258 @@
+"""Implementation of Rule L068."""
+
+from typing import List, Optional
+
+from sqlfluff.core.parser import WhitespaceSegment
+
+from sqlfluff.core.parser import BaseSegment
+from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_fix_compatible,
+    document_groups,
+)
+import sqlfluff.core.rules.functional.segment_predicates as sp
+
+
+@document_groups
+@document_configuration
+@document_fix_compatible
+class Rule_L068(BaseRule):
+    """The brackets should stand at the same line with from/as/join keywords.
+    The ending brackets followed by as keyword should stand at the beginning
+    of a new line.
+
+    **Anti-pattern**
+
+    Opening bracket following from keyword takes a new line,
+    and as clause at the end takes a new line instead of following end bracket
+
+    .. code-block:: sql
+        :force:
+
+        SELECT
+            A.SOMETHING
+        FROM
+            (SELECT B FROM D GROUP BY 1)
+        AS A
+
+
+
+    **Best practice**
+
+    Opening bracket following after from keywords with no newline, and ending bracket
+    followed by as clause with no newline.
+
+    .. code-block:: sql
+        :force:
+
+        SELECT
+            A.SOMETHING
+        FROM (SELECT B FROM D GROUP BY 1) AS A
+
+    """
+    groups = ("all",)
+
```

Comment:
(same as my comment on #3726 ).

Now that I've merged #3717 you'll need to add a `crawl_behaviour` attribute to govern how this rule crawls.

Looking at what you've done, it might be a bit more complicated in this case.

To pick up your conditions on `if context.segment.is_type('from_clause')` and `if context.segment.is_type('join_clause')`, you could very easily just set `crawl_behaviour = SegmentSeekerCrawler({"from_clause", "join_clause"})`.

The tricky one will be `if children.any(sp.is_type('alias_expression'))`. In this case I think you have a few options:
1. Given you're looking for bracketed expressions followed by an alias, you could also add `bracketed` to your search query and then use `parent_stack` to go up one level and see if the `bracketed` is followed by an `alias_expression`.
2. You could find the instances where aliases might follow a table expression, which I think involves only adding `select_clause_element` to your query (to catch examples like `SELECT (SELECT 1) as bar`).

My gut says the latter would be less effort. Let me know if you need a had reworking this - I realise this is new functionality, but it should drastically increase performance for rules exactly like this one.